src/main/java/com/amazon/ion/impl/bin/IonManagedWriter_1_1.kt

```diff
@@ -316,20 +316,132 @@ internal class IonManagedWriter_1_1(
     }
 
     override fun writeIonVersionMarker() {
-        // Make sure we write out any symbol tables and buffered values before the IVM
-        flush()
-        userData.writeIVM()
+        if (depth == 0) {
+            // Make sure we write out any symbol tables and buffered values before the IVM
+            finish()
+            systemData.writeIVM()
+        } else {
+            writeSymbol("\$ion_1_1")
```

Comment:
This isn't an IVM... should this be an error? Or are we just saying that it's not the caller's responsibility know whether a real IVM is legal in this position?

Reply:
I thought it should be an error (which is why I originally implemented it the way I did), but then I read the Javadoc for the inherited method.

> Write an Ion version marker symbol to the output. This is the $ion_1_0 value currently (in later versions the number may change). In text output this appears as the text symbol. In binary this will be the symbol id if the writer is in a list, sexp or struct. If the writer is currently at the top level this will write the "magic cookie" value. Writing a version marker will reset the symbol table to be the system symbol table.

This seems pretty clear that it should just write a symbol if the reader is in any container.